src/status_im/common/lightbox/top_view.cljs

```diff
@@ -40,33 +39,14 @@
         (anim/animate top-view-width screen-width)
         (anim/animate top-view-bg colors/neutral-100-opa-0)))))
 
-(defn drawer
-  [images index]
-  (let [{:keys [image]} (nth images index)
-        uri             (url/replace-port image (rf/sub [:mediaserver/port]))]
-    [quo/action-drawer
-     [[{:icon                :i/save
-        :accessibility-label :save-image
-        :label               (i18n/label :t/save-image-library)
-        :on-press            (fn []
-                               (rf/dispatch [:hide-bottom-sheet])
-                               (rf/dispatch
-                                [:lightbox/save-image-to-gallery
-                                 uri
-                                 #(rf/dispatch [:toasts/upsert
-                                                {:id              :random-id
-                                                 :type            :positive
-                                                 :container-style {:bottom (when platform/android? 20)}
-                                                 :text            (i18n/label :t/photo-saved)}])]))}]]]))
-
 (defn share-image
   [images index]
   (let [{:keys [image]} (nth images index)
         uri             (url/replace-port image (rf/sub [:mediaserver/port]))]
     (rf/dispatch [:lightbox/share-image uri])))
 
 (defn top-view
-  [images insets index animations derived landscape? screen-width]
+  [images insets index animations derived landscape? screen-width options-drawer-component]
```

Comment:
imo this is cleaner if we just pass the `on-press` instead of the `options-drawer-component`

Reply:
Agree. I updated the code to address this